src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -310,7 +310,7 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
                 forbidden.remove(prop)
 
     # set default storage SKU if not provided and using an image based OS
-    if not namespace.storage_sku and namespace.storage_profile not in [StorageProfile.ManagedSpecializedOSDisk, StorageProfile.SASpecializedOSDisk]:  # pylint: disable=line-too-long
+    if not namespace.storage_sku and namespace.storage_profile in [StorageProfile.SAPirImage, StorageProfile.SACustomImage]:  # pylint: disable=line-too-long
         namespace.storage_sku = 'Standard_LRS' if for_scale_set else 'Premium_LRS'
```

Comment:
Yep, i got that point. I meant we double check, if it becomes `premium_lrs` by default, then we should communicate to CRP folks, as the costing change will be more significant for scale-set than loose vm